cdap-examples/LogAnalysis/src/main/java/co/cask/cdap/examples/loganalysis/LogAnalysisApp.java

```diff
@@ -165,4 +177,80 @@ public void centers(HttpServiceRequest request, HttpServiceResponder responder,
       }
     }
   }
+
+  /**
+   * A Service which serves the number of requests made by unique ip address from a {@link TimePartitionedFileSet}
+   */
+  public static final class RequestCounterHanlder extends AbstractHttpServiceHandler {
+
+    private static final Gson GSON = new Gson();
+    static final String REQUEST_COUNTER_PARTITIONS_HANDLER = "reqcount";
+    static final String REQUEST_FILE_CONTENT_HANDLER = "reqfile";
+    static final String REQUEST_FILE_PATH_HANDLER_KEY = "time";
+
+    @UseDataSet(REQ_COUNT_STORE)
+    private TimePartitionedFileSet reqCountStore;
+
+    /**
+     * Handler which lists all the different time partitions available in the {@link LogAnalysisApp#REQ_COUNT_STORE}
+     * {@link TimePartitionedFileSet}
+     */
+    @Path(REQUEST_COUNTER_PARTITIONS_HANDLER)
+    @GET
+    public void getRequestFilesetPartitions(HttpServiceRequest request, HttpServiceResponder responder) {
+
+      // get all the existing paritions
+      Set<TimePartitionDetail> partitionsByTime = reqCountStore.getPartitionsByTime(0, Long.MAX_VALUE);
+
+      if (partitionsByTime.isEmpty()) {
+        responder.sendString(HttpURLConnection.HTTP_NO_CONTENT, "No request count information found", Charsets.UTF_8);
+      } else {
+        Set<String> formatedTimes = Sets.newHashSet();
```

Comment:
Should this somehow be sorted by date? I think with HashSet they will be in random order...